ros2doctor/ros2doctor/api/package.py

```diff
@@ -97,43 +97,51 @@ def compare_versions(result: Result, local_packages: dict, distro_packages: dict
     :param: boolean value determines which output to populate, msgs or report
     :return: list of warning messages
     """
-    missing_req = ''
-    missing_local = ''
+    missing_req = []
+    missing_local = []
     for name, local_ver_str in local_packages.items():
         if not local_ver_str:
-            missing_local += ' ' + name
+            missing_local += name
             local_ver_str = ''
```

Comment:
I know you didn't add this, but there doesn't seem to be a reason for this line of code.  We already know that the local_ver_str is empty by this point.

```suggestion
```